presto-resource-group-managers/src/main/java/io/prestosql/plugin/resourcegroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -329,21 +334,23 @@ private synchronized void populateFromDbHelper(Map<Long, ResourceGroupSpecBuilde
         return new AbstractMap.SimpleImmutableEntry<>(managerSpec, resourceGroupSpecs);
     }
 
-    private synchronized void configureChangedGroups(Set<ResourceGroupIdTemplate> changedSpecs)
+    @GuardedBy("this")
+    private void configureChangedGroups(Set<ResourceGroupIdTemplate> changedSpecs)
     {
         for (ResourceGroupIdTemplate resourceGroupIdTemplate : changedSpecs) {
-            for (ResourceGroupId resourceGroupId : configuredGroups.getOrDefault(resourceGroupIdTemplate, ImmutableList.of())) {
+            for (ResourceGroupId resourceGroupId : configuredGroups.get(resourceGroupIdTemplate)) {
                 synchronized (getRootGroup(resourceGroupId)) {
                     configureGroup(groups.get(resourceGroupId), resourceGroupSpecs.get(resourceGroupIdTemplate));
                 }
             }
         }
     }
 
-    private synchronized void disableDeletedGroups(Set<ResourceGroupIdTemplate> deletedSpecs)
+    @GuardedBy("this")
+    private void disableDeletedGroups(Set<ResourceGroupIdTemplate> deletedSpecs)
```

Comment:
Isn't `@GuardedBy("this")` an equivalent of `synchronized` method modifier? If so then method should be sill `synchronized`.

Reply:
Technically -- yes, this could stay `synchronized`.
However, if we have `configureChangedGroups`, `disableDeletedGroups` methods as `synchronized`, this suggests we can call them without synchronization. 
However, `configuredGroups` tracks what needs to be reconfigured. If we allow adding to this set and reading without any other synchronization, the collection is synchronized correctly, but the information flow is not. We would be able to miss some "need reconfigure" or "need delete" information.